SHARE-491 Add Share function to User Profile Screen

Kropiunig opened the pull request:
Added a Share button next to the search button in the topbar of the User Profile screen.

When pressed the user is able to share a link to a profile

---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [x] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no warnings and errors 
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Verify that all tests are passing (CI), if not please state the test cases in the [section](#Tests) below
- [x] Perform a self-review of the changes
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [x] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer
- [x] Check that your pull request has been successfully deployed to https://web-test-1.catrob.at/

### Additional Description
`TODO: Add additional information that is not in your commit-message here`

### Tests - additional information
`TODO: add additional information about testruns here`

Kropiunig committed:
SHARE-491 Add Share function to User Profile Screen

Kropiunig committed:
SHARE-491 Add Share function to User Profile Screen

dmetzner commented on assets/js/profile.js:
```diff
@@ -5,9 +5,23 @@ import 'external-svg-loader'
 require('../styles/components/achievements.scss')
 
 const programs = new ProjectLoader('#user-programs', apiUserPrograms)
+const $projectShare = $('.js-user-share')
```
$projectShare name for user-share does not fit

dmetzner commented on templates/UserManagement/Profile/profile.html.twig:
```diff
@@ -175,6 +184,17 @@
 
+  <div class="js-user-share"
+       data-theme-display-name="{{ getThemeDisplayName() }}"
+       data-trans-check-out-project="{{ "checkOutProject"|trans({}, "catroweb") }}"
```
"Checkout project" seems not to fit the current context. Please add a new translation.

dmetzner reviewed (changes requested):
Please also add a simple test so that the feature is not lost in a year or two.

To start maybe take a look at: tests/behat/features/web/top-bar/top_app_bar_more_options.feature


Not necessary to be done in this PR, however, it would be great if the JS is refactored to use the share functionality independent from its context. In other words, it would be great if we only have one JS class that can be easily used to share a project, a user, or something else in the future. Lets' make the implementation a bit more general.

Kropiunig committed:
SHARE-491 Add Share function to User Profile Screen

added share functionality to userprofile and wrote test

Kropiunig committed:
SHARE-491 Add Share function to User Profile Screen

Added a Share button next to the search button in the topbar of the User Profile screen.

When pressed the user is able to share a link to a profile

dmetzner commented on assets/js/profile.js:
```diff
@@ -1,12 +1,28 @@
+shareLink(
+  $userShare.data('theme-display-name'),
+  $userShare.data('trans-check-out-project'),
+  $userShare.data('project-url'),
```
data-user-url

dmetzner replied:
![image](https://user-images.githubusercontent.com/40868718/156708808-9b520c0a-8062-442b-99ca-35cbfbc6b241.png)

dmetzner commented on templates/UserManagement/Profile/profile.html.twig:
```diff
@@ -1,10 +1,19 @@
+
+    <meta property="og:image" content="{{ asset(profile.avatar) }}"/>
+    <meta property="og:type" content="website"/>
+    <meta property="og:image:secure_url" content="{{ asset(profile.avatar) }}"/>
```
ich glaube das wird momentan nicht funktionieren weil es ka keine URL ist sondern ein base64 Bild... Aber fürs erste können wir es ruhig so lassen.

dmetzner reviewed (changes requested):
Bitte noch den user url fixen :)

Kropiunig committed:
SHARE-491 Add Share function to User Profile Screen

Added a Share button next to the search button in the topbar of the User Profile screen.

When pressed the user is able to share a link to a profile
